concrete-core/src/backends/core/implementation/engines/ggsw_ciphertext_conversion.rs

```diff
@@ -0,0 +1,200 @@
+    /// let glwe_dimension = GlweDimension(2);
+    /// let polynomial_size = PolynomialSize(256);
+    /// let level = DecompositionLevelCount(1);
+    /// let base_log = DecompositionBaseLog(4);
+    /// // Here a hard-set encoding is applied (shift by 50 bits)
+    /// let input = 3_u64 << 50;
+    /// let noise = Variance(2_f64.powf(-25.));
+    ///
+    /// let mut engine = CoreEngine::new()?;
+    /// let key: GlweSecretKey64 = engine.create_glwe_secret_key(glwe_dimension, polynomial_size)?;
+    /// let plaintext = engine.create_plaintext(&input)?;
+    ///
+    /// let ciphertext = engine.encrypt_ggsw_ciphertext(&key, &plaintext, noise, level, base_log)?;
+    ///
+    /// let fourier_ciphertext: GgswCiphertextComplex64 =
+    ///     engine.convert_ggsw_ciphertext(&ciphertext)?;
+    /// #
+    /// assert_eq!(fourier_ciphertext.glwe_dimension(), glwe_dimension);
+    /// assert_eq!(fourier_ciphertext.polynomial_size(), polynomial_size);
+    /// assert_eq!(fourier_ciphertext.decomposition_base_log(), base_log);
+    /// assert_eq!(fourier_ciphertext.decomposition_level_count(), level);
+    ///
+    /// engine.destroy(key)?;
+    /// engine.destroy(plaintext)?;
+    /// engine.destroy(ciphertext)?;
+    /// engine.destroy(fourier_ciphertext)?;
+    /// #
+    /// # Ok(())
+    /// # }
+    /// ```
+    fn convert_ggsw_ciphertext(
+        &mut self,
+        input: &GgswCiphertext64,
+    ) -> Result<GgswCiphertextComplex64, GgswCiphertextConversionError<Self::EngineError>> {
+        Ok(unsafe { self.convert_ggsw_ciphertext_unchecked(input) })
+    }
+
+    unsafe fn convert_ggsw_ciphertext_unchecked(
+        &mut self,
+        input: &GgswCiphertext64,
+    ) -> GgswCiphertextComplex64 {
+        let input_bsk = StandardBootstrapKey::from_container(
+            input.0.as_tensor().as_container().clone(),
+            input.glwe_dimension().to_glwe_size(),
+            input.polynomial_size(),
+            input.decomposition_level_count(),
+            input.decomposition_base_log(),
+        );
+        let output = ImplFourierBootstrapKey::allocate(
+            Complex64::new(0., 0.),
+            input.glwe_dimension().to_glwe_size(),
+            input.polynomial_size(),
+            input.decomposition_level_count(),
+            input.decomposition_base_log(),
+            LweDimension(1),
+        );
+        let mut output_bsk = FourierLweBootstrapKey64(output);
+        let buffers = self.get_fourier_bootstrap_u64_buffer(&output_bsk);
+        output_bsk.0.fill_with_forward_fourier(&input_bsk, buffers);
+        let output = GgswCiphertext::from_container(
+            output_bsk.0.as_tensor().as_container().clone(),
+            output_bsk.glwe_dimension().to_glwe_size(),
+            output_bsk.polynomial_size(),
+            output_bsk.decomposition_base_log(),
+        );
+        GgswCiphertextComplex64(output)
+    }
+}
+
+impl<Ciphertext> GgswCiphertextConversionEngine<Ciphertext, Ciphertext> for CoreEngine
```

Comment:
Here I do not understand why we need this?

Reply:
Honestly, I'm not sure. To create this failed I copied another *_conversion.rs and a similar function was defined. I assumed it was some kind of failsafe.

Reply:
@aPere3 might have some insight.

Reply:
@aPere3 where should we do that?

Reply:
@damienligier I opened an issue so we know we have to add this comment in a further PR.

Reply:
I don't think there is much to say, but yes let's write a little something on top of the impl block if that can make others more comfortable.

Reply:
@aPere3 we could just write what you explained just above:
```
This blanket implementation allows to convert from a type to itself by just cloning the value. It is important to have it for the completeness of the genericity.
```

Reply:
If its a doc `///` the first sentence is enough I think.

Reply:
I'll add it to my *_conversion.rs but I won't check the whole code for similar impl, that PR is big enough already.

Reply:
Sure!